### PR TITLE
The Blog App: Controllers Specs

### PR DESCRIPTION
**Pull request by romans-adi**
This PR introduces a new set of request specs to thoroughly test the `UsersController` and `PostsController` in the application.

## Testing Approach
- Utilized `RSpec` for request testing.
- Employed `subject` for concise and clear test cases.
- Ensured that the response status and rendered templates were appropriately validated.
- Checked the response bodies for the presence of expected content.

## UsersController
- Implemented tests for the `GET #index` and `GET #show` actions.
- Utilized a sample `user` instance with mock attributes to simulate a user.
- Verified the success of the action's response.
- Ensured that the corresponding show template is rendered.
- Checked for the inclusion of the user's name in the response body.

## PostsController
- Implemented tests for the `GET #index` and `GET #show` actions.
- Generated a mock `user` and a related `post` instance with simulated data.
- Verified the success of the action's response.
- Confirmed the rendering of the associated show template.

### CODE REVIEW

File: spec/requests/controllers_spec.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    describe 'GET #show' do
+      subject { get user_post_path(user, post) }
+
+      it 'returns a successful response' do
+        subject
+        expect(response).to be_successful
+      end
+
+      it 'renders the show template' do
```
Comment (Davidosky007) [inline spec/requests/controllers_spec.rb:97]: * Kindly write a test to check `If the response body includes correct placeholder text.` as this is missing and it's [required](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=If%20the%20response%20body%20includes%20correct%20placeholder%20text.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @romans-adi 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* No linter errors 👍 
* All tests passed 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/controllers_spec.rb (hunk 1 above)
Comment (romans-adi) [reply spec/requests/controllers_spec.rb:97]: Thank you, @Davidosky007, you're absolutely correct! I overlooked this part and forgot to include it.

**Review by Munsa1 — APPROVED** [review]
Hi @romans-adi, 
congrats 2

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Required Change :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**